### ISSUE TITLE
Splitter crash on a corrupt stream

### ISSUE THREAD

**Issue by trink** [mozilla-services/heka#1381]
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x50 pc=0x4cbada]

goroutine 1 [running]:
github.com/mozilla-services/heka/pipeline.(*HekaFramingSplitter).FindRecord(0xc20801e4c0, 0xc208075d20, 0x2e0, 0x2e0, 0x2, 0x0, 0x0, 0x0)
	/work/git/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/splitters.go:224 +0x1ba
github.com/mozilla-services/heka/pipeline.(*sRunner).GetRecordFromStream(0xc208032370, 0x7f47eb23eb08, 0xc208044068, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0)
	/work/git/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/splitter_runner.go:195 +0x265
main.main()
	/work/git/heka/build/heka/src/github.com/mozilla-services/heka/cmd/heka-cat/main.go:113 +0xadc

goroutine 5 [syscall]:
```

**Comment by rafrombrc** [comment]
This might be resolved by #1375. In any event, it's definitely the same issue, the splitter is trying to use its SplitterRunner to log the error but none has been set. Can you test #1375 to see if this resolves?

**Comment by trink** [comment]
For some reason I thought that already landed in dev, yes that fixes the issue.
